test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/ccm/BaseCcmRule.java

```diff
@@ -59,24 +57,27 @@ protected void after() {
 
   @Override
   public Statement apply(Statement base, Description description) {
+    // If test is annotated with CassandraRequirement or DseRequirement, ensure configured CCM
+    // cluster meets those requirements.
     CassandraRequirement cassandraRequirement =
         description.getAnnotation(CassandraRequirement.class);
 
     if (cassandraRequirement != null) {
       // if the configured cassandra cassandraRequirement exceeds the one being used skip this test.
       if (!cassandraRequirement.min().isEmpty()) {
         CassandraVersion minVersion = CassandraVersion.parse(cassandraRequirement.min());
-        if (minVersion.compareTo(cassandraVersion) > 0) {
-          // Create a statement which simply indicates that the configured cassandra cassandraRequirement is too old for this test.
+        if (minVersion.compareTo(ccmBridge.getCassandraVersion()) > 0) {
+          // Create a statement which simply indicates that the configured cassandra
+          // cassandraRequirement is too old for this test.
           return new Statement() {
 
             @Override
-            public void evaluate() throws Throwable {
+            public void evaluate() {
               throw new AssumptionViolatedException(
                   "Test requires C* "
                       + minVersion
                       + " but "
-                      + cassandraVersion
+                      + ccmBridge.getCassandraVersion()
                       + " is configured.  Description: "
                       + cassandraRequirement.description());
             }
```

Comment:
This is repeated 4 times, maybe we should extract a method that creates the Statement.

Reply:
Good idea, this code is a big repetitive.